components/org.wso2.analytics.is/src/main/test/org/wso2/analytics/is/siddhi/extension/JsonFlatMapEventDuplicatorStreamProcessorTestCase.java

```diff
@@ -0,0 +1,139 @@
+                        Assert.assertEquals("tester.com", inEvents[0].getData(2));
+                        Assert.assertEquals("{\"urn:scim:schemas:core:1.0:userName\":\"john\"," +
+                                        "\"http://wso2.org/claims/organization\":\"IBM\"," +
+                                        "\"http://wso2.org/claims/emailaddress\":\"john@gmail.com\"}",
+                                inEvents[0].getData(3));
+                        Assert.assertEquals("urn:scim:schemas:core:1.0:userName", inEvents[0].getData(4));
+                        Assert.assertEquals("john", inEvents[0].getData(5));
+
+                        Assert.assertEquals("john", inEvents[1].getData(0));
+                        Assert.assertEquals("LEGACY", inEvents[1].getData(1));
+                        Assert.assertEquals("tester.com", inEvents[1].getData(2));
+                        Assert.assertEquals("{\"urn:scim:schemas:core:1.0:userName\":\"john\"," +
+                                        "\"http://wso2.org/claims/organization\":\"IBM\"," +
+                                        "\"http://wso2.org/claims/emailaddress\":\"john@gmail.com\"}",
+                                inEvents[1].getData(3));
+                        Assert.assertEquals("http://wso2.org/claims/organization", inEvents[1].getData(4));
+                        Assert.assertEquals("IBM", inEvents[1].getData(5));
+
+                        Assert.assertEquals("john", inEvents[2].getData(0));
+                        Assert.assertEquals("LEGACY", inEvents[2].getData(1));
+                        Assert.assertEquals("tester.com", inEvents[2].getData(2));
+                        Assert.assertEquals("{\"urn:scim:schemas:core:1.0:userName\":\"john\"," +
+                                        "\"http://wso2.org/claims/organization\":\"IBM\"," +
+                                        "\"http://wso2.org/claims/emailaddress\":\"john@gmail.com\"}",
+                                inEvents[2].getData(3));
+                        Assert.assertEquals("http://wso2.org/claims/emailaddress", inEvents[2].getData(4));
+                        Assert.assertEquals("john@gmail.com", inEvents[2].getData(5));
+                    }
+                }
+            }
+        });
+
+        InputHandler inputHandler = executionPlanRuntime.getInputHandler("claimsStream");
+        executionPlanRuntime.start();
+        inputHandler.send(new Object[]{"nadund", "PRIMARY", "carbon.super",
+                "{\"urn:scim:schemas:core:1.0:userName\":\"nadund\"," +
+                        "\"http://wso2.org/claims/organization\":\"WSO2\"}"});
+        inputHandler.send(new Object[]{"john", "LEGACY", "tester.com",
+                "{\"urn:scim:schemas:core:1.0:userName\":\"john\"," +
+                        "\"http://wso2.org/claims/organization\":\"IBM\"," +
+                        "\"http://wso2.org/claims/emailaddress\":\"john@gmail.com\"}"});
+        Thread.sleep(4000);
```

Comment:
Normally it is not recommended to add thread.sleep in test cases, without a condition. If you are waiting for some event to occur shall we add a loop to check whether it is satisfied and add this as timeout value if possible

Reply:
Generally, in siddhi extension test cases, they use Thread.sleep(). We anyway have to wait for sometime to make sure that it only sends exactly the required number of events out. If we add a check to wait until the first arrived event we can't guarantee that it's not sending more than the expected number of events. So generally we should wait.